### PR TITLE
Red Comrades Engine

### PR DESCRIPTION
**Pull request by whiterandrek**
This PR adds support for the Red Comrades 1, Red Comrades 2 and Red Comrades Demo by S.K.I.F
The demo is fully supported.
The other games can run, but are unplayable due to a lot of bugs.

The engine misses walking system implementation. This is the main reason why the engine is bugged.

I'm not sure if it should be merged in current state.
Looking forward to your suggestions.

### CODE REVIEW

File: engines/petka/configure.engine [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# This file is included from the main "configure" script
+# add_engine [name] [desc] [build-by-default] [subengines] [base games] [deps]
+add_engine petka "Red Comrades" no "" "" "highres"
```
Comment (ccawley2011) [inline engines/petka/configure.engine]:
```suggestion
add_engine petka "Red Comrades" no "" "" "highres 16bit"
```

File: engines/petka/sound.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+void SoundMgr::removeAll() {
+	debug("SoundMgr::removeAll");
+	_sounds.clear(0);
+}
+
+void SoundMgr::removeSoundsWithType(Audio::Mixer::SoundType type) {
+	SoundsMap::iterator it;
+	for (it = _sounds.begin(); it != _sounds.end(); ++it) {
```
Comment (winterheart) [inline engines/petka/sound.cpp]: this should be rewrote into
```
for (it = _sounds.begin(); it != _sounds.end();) {
  Sound *s = it->_value.get();
  if (s->type() == type && !s->isPlaying()) {
    it = _sounds.erase(it);
  } else {
    it++;
  }
}
```
or else whole cycle run into SIGSERV on deleted iterator value

**Comment by Thunderforge** [comment]
I think something went wrong with this PR (a failed rebase?). There are currently 5,807 commits, and many of them don't have anything to do with this engine. Can that get resolved?

**Comment by sev-** [comment]
The last push is all wrong. You have to rebase it. You got a bunch of irrelevant commits in there.

**Pull request merged**